### ISSUE TITLE
Cpp extensions do not work

### ISSUE THREAD

**Issue by AAlguacil** [jolibrain/fluidnet_cxx#7]
At execution, in the function [Advect Scalar](https://github.com/jolibrain/fluidnet_cxx/blob/3542f19f383cd5eb5d1d097744a14de7c3ec1b48/pytorch/lib/fluid/cpp/fluids_init.cpp#L257), called from [advection.py](https://github.com/jolibrain/fluidnet_cxx/blob/3542f19f383cd5eb5d1d097744a14de7c3ec1b48/pytorch/lib/fluid/cpp/advection.py#L64) input arguments behave as `Variable` type (grad is computed) whereas intermediate objects in the function are just `Tensor` type.

```
at::Tensor advectScalar
(
  float dt, T src, T U, T flags,
  const std::string method_str,
  int bnd,
  const bool sample_outside_fluid,
  const float maccormack_strength
) {
  // Size checking done in python side
  int bsz = flags.size(0);
  int d = flags.size(2);
  int h = flags.size(3);
  int w = flags.size(4);

  bool is3D = (U.size(1) == 3);

  src.set_requires_grad(false);
  U.set_requires_grad(false);
  flags.set_requires_grad(false);

  T s_dst = zeros_like(src);

  T fwd = zeros_like(src);
  T bwd = zeros_like(src);
  T fwd_pos = zeros_like(U);
  T bwd_pos = zeros_like(U);

  std::cout << "Is variable?" << std::endl;
  std::cout << src.is_variable() << std::endl;
  // returns 1, should be 0
  std::cout << U.is_variable() << std::endl;
  // returns 1, should be 0
  std::cout << flags.is_variable() << std::endl;
  // returns 1, should be 0

```

All should be of `Tensor` type as no backpropagation is required in the advection part.

**Comment by AAlguacil** [comment]
> At execution, in the function [Advect Scalar](https://github.com/jolibrain/fluidnet_cxx/blob/3542f19f383cd5eb5d1d097744a14de7c3ec1b48/pytorch/lib/fluid/cpp/fluids_init.cpp#L257), called from [advection.py](https://github.com/jolibrain/fluidnet_cxx/blob/3542f19f383cd5eb5d1d097744a14de7c3ec1b48/pytorch/lib/fluid/cpp/advection.py#L64) input arguments behave as `Variable` type (grad is computed) whereas intermediate objects in the function are just `Tensor` type.
> 
> ```
> at::Tensor advectScalar
> (
>   float dt, T src, T U, T flags,
>   const std::string method_str,
>   int bnd,
>   const bool sample_outside_fluid,
>   const float maccormack_strength
> ) {
>   // Size checking done in python side
>   int bsz = flags.size(0);
>   int d = flags.size(2);
>   int h = flags.size(3);
>   int w = flags.size(4);
> 
>   bool is3D = (U.size(1) == 3);
> 
>   src.set_requires_grad(false);
>   U.set_requires_grad(false);
>   flags.set_requires_grad(false);
> 
>   T s_dst = zeros_like(src);
> 
>   T fwd = zeros_like(src);
>   T bwd = zeros_like(src);
>   T fwd_pos = zeros_like(U);
>   T bwd_pos = zeros_like(U);
> 
>   std::cout << "Is variable?" << std::endl;
>   std::cout << src.is_variable() << std::endl;
>   // returns 1, should be 0
>   std::cout << U.is_variable() << std::endl;
>   // returns 1, should be 0
>   std::cout << flags.is_variable() << std::endl;
>   // returns 1, should be 0
> ```
> 
> All should be of `Tensor` type as no backpropagation is required in the advection part.

@beniz Hello! Could you have a look at this, as I am a bit stucked trying to install fluidnet with pytorch 1.0.

**Comment by AAlguacil** [comment]
Closed  `[ a2a49ac](https://github.com/jolibrain/fluidnet_cxx/commit/a2a49acec6d54c037fae57b437a55bb5142346ed)`
